src/simtools/ray_tracing/psf_parameter_optimisation.py

```diff
@@ -0,0 +1,794 @@
+    # tel_model.change_multiple_parameters(**pars_to_change)
+
+    pars = {
+        "mirror_reflection_random_angle": [
+            mirror_reflection,
+            mirror_reflection_fraction,
+            mirror_reflection_2,
+        ],
+        "mirror_align_random_horizontal": [mirror_align, 28.0, 0.0, 0.0],
+        "mirror_align_random_vertical": [mirror_align, 28.0, 0.0, 0.0],
+    }
+    all_parameters.append(pars)
+
+
+def get_previous_values(tel_model):
+    """
+    Retrieve previous parameter values from the telescope model.
+
+    Parameters
+    ----------
+    tel_model : TelescopeModel
+        Telescope model object.
+
+    Returns
+    -------
+    tuple
+        Tuple containing the previous values of mirror_reflection_random_angle (first entry),
+        mirror_reflection_fraction, second entry), mirror_reflection_random_angle (third entry),
+        and mirror_align_random_horizontal/vertical.
+    """
+    split_par = tel_model.get_parameter_value("mirror_reflection_random_angle")
+    mrra_0, mfr_0, mrra2_0 = split_par[0], split_par[1], split_par[2]
+    mar_0 = tel_model.get_parameter_value("mirror_align_random_horizontal")[0]
+    logger.debug(
+        "Previous parameter values:\n"
+        f"MRRA = {mrra_0!s}\n"
+        f"MRF = {mfr_0!s}\n"
+        f"MRRA2 = {mrra2_0!s}\n"
+        f"MAR = {mar_0!s}\n"
+    )
+    return mrra_0, mfr_0, mrra2_0, mar_0
+
+
+def generate_random_parameters(
+    all_parameters, n_runs, args_dict, mrra_0, mfr_0, mrra2_0, mar_0, tel_model
+):
+    """
+    Generate random parameters for tuning.
+
+    Parameters
+    ----------
+    all_parameters : list
+        List to store all parameter sets.
+    n_runs : int
+        Number of random parameter combinations to test.
+    args_dict : dict
+        Dictionary containing parsed command-line arguments.
+    mrra_0 : float
+        Initial value of mirror_reflection_random_angle.
+    mfr_0 : float
+        Initial value of mirror_reflection_fraction.
+    mrra2_0 : float
+        Initial value of the second mirror_reflection_random_angle.
+    mar_0 : float
+        Initial value of mirror_align_random_horizontal/vertical.
+    tel_model : TelescopeModel
+        Telescope model object to check if it's a dual mirror telescope.
+    """
+    # Range around the previous values are hardcoded
+    # Number of runs is hardcoded
+    if args_dict["fixed"]:
+        logger.debug("fixed=True - First entry of mirror_reflection_random_angle is kept fixed.")
+
+    # Check if telescope is dual mirror and set mar to 0 if so
+    is_dual_mirror = model_utils.is_two_mirror_telescope(tel_model.name)
+    if is_dual_mirror:
+        mar_fixed_value = 0.0
+    else:
+        mar_fixed_value = None
+
+    for _ in range(n_runs):
+        mrra_range = 0.004 if not args_dict["fixed"] else 0
+        mrf_range = 0.1
+        mrra2_range = 0.03
+        mar_range = 0.005
+        rng = np.random.default_rng(seed=100)  # Fixed seed for reproducibility
```

Comment:
Trying to understand the fixed seed. We usually do this only for testing, but we require 'true' randomness for full runs. Isn't otherwise the danger that all our results are biased in a certain way? Just think that this seeds gives you by chance a large upwards fluctuation. I suggest to remove it (or make it configurable)